pkg/utils/injection/injection.go

```diff
@@ -79,3 +79,17 @@ func GetControllerName(ctx context.Context) string {
 	}
 	return name.(string)
 }
+
+type testInstanceKey struct{}
```

Comment:
This injection pattern is useful to keep interfaces in production code clean. I'm wary of coupling test implementation details to this injection package. There are some other examples of injecting data into the EC2 fake. Happy to help offline if you need.

Reply:
Yeah this may not be ideal. I am removing it and implements a function to switch on and off those special instance types instead.